### PR TITLE
Predictive changes start

### PR DESCRIPTION
**Pull request by thatch45**
I have started to add the predictive changes key to the state return data structure. This is all slated for Boron and we will want to get some involvement to fill this out. I have only started on the file state module

### CODE REVIEW

**Comment by thatch45** [comment]
Start on #24938

File: doc/ref/states/writing.rst [hunk 1]
```diff
@@ -97,6 +97,10 @@ A State Module must return a dict containing the following keys/values:
   *False*.
 - **comment:** A string containing a summary of the result.
 
+The return data can also, include the **pchanges** key, this statnds for
```
Comment (bechtoldt) [inline doc/ref/states/writing.rst]: **stands**

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
@thatch45 I have submitted a PR against your fork to fix the tests. After you rebase this and merge those, we should be good to get this in.

**Comment by cachedout** [comment]
Closed in favor of #26134

**Pull request closed without merging**